### PR TITLE
[v6r21] Add Docker Compose support for development purpose.

### PR DESCRIPTION
**Pull request by pujanm**
This will add Docker Compose setup for DIRAC Development environment.
This sets up DIRAC, MySQL and ElasticSearch on running a few commands.
This setup is very helpful for container-based development for DIRAC.

### CODE REVIEW

**Comment by fstagni** [comment]
Thanks for this. I am not familiar with docker-compose, so can't really comment technically. I will try to find the time to try this out. This has a high potential to be added to this guide: https://dirac.readthedocs.io/en/latest/DeveloperGuide/DevelopmentEnvironment/DeveloperInstallation/stuffThatRun.html 
Would you have a look into how that guide can be expanded?

File: container/docker_compose_setup/dockerfiles/MySQL_Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM mysql/mysql-server:5.7
```
Comment (fstagni) [inline container/docker_compose_setup/dockerfiles/MySQL_Dockerfile]: I think we may rename this file "MySQL57_Dockerfile", and add a second named "MySQL80_Dockerfile" for the MySQL 8.0 version. 
Similar can be said for the "DIRAC_Dockerfile" and slc6 vs cc7.

Comment (pujanm) [reply container/docker_compose_setup/dockerfiles/MySQL_Dockerfile]: Sure will make these changes.

**Comment by chaen** [comment]
Thanks for that.
From a quick glance, it seems to me that the content of many of your bash scripts overlaps with the one of the tutorial for setting up the machine (https://github.com/DIRACGrid/DIRAC/blob/integration/docs/source/AdministratorGuide/Tutorials/basicTutoSetup.sh) 
Would there be a way to reuse that to avoid duplicates ?

**Comment by pujanm** [comment]
@chaen I actually referred this tutorial for developing the bash scripts: https://dirac.readthedocs.io/en/latest/AdministratorGuide/Tutorials/basicTutoSetup.html
Didn't knew such bash script was already there.
We can try to reuse the one you pointed but I developed the script keeping Docker in mind.

File: container/docker_compose_setup/env_files/env_dirac.cfg [hunk 2]
```diff
@@ -0,0 +1,8 @@
+DIRAC
+{
+  Setup = DIRAC-Devlopment
```
Comment (atsareg) [inline container/docker_compose_setup/env_files/env_dirac.cfg]: Developement

**Comment by atsareg** [comment]
This is a very interesting example of yet another way of the DIRAC service installation that I am keen to try out on my Mac. Since this is completely complementary to other codes, I suggest to merge it straight away unless there are real serious objections - any ?

**Comment by fstagni** [comment]
It does not necessarily need to be part of any release. It's being tried out by our students, it might get an update soon.

**Pull request merged**